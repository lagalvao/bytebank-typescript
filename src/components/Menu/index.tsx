const menuList = [
  'Inicial',
  'Transferências',
  'Investimentos',
  'Outros serviços',
]

export default function Menu() {
  return (
    <nav className="w-[180px] h-full p-6 bg-slate-100  list-none rounded-lg">
      {menuList.map((item, index) => {
        return (
          <div key={item} className="text-center">
            <a
              href="/"
              className="text-slate-900 text-base cursor-pointer hover:text-black font-semibold"
            >
              {item}
            </a>
            {index !== menuList.length - 1 && (
              <div className="w-[90%] h-[1px] bg-slate-900 mx-auto my-4" />
            )}
          </div>
        )
      })}
    </nav>
  )
}
