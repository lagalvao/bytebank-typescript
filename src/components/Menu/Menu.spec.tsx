import { render, screen } from '@testing-library/react'
import Menu from '.'

describe('Menu', () => {
  it('should render a link to the home page', () => {
    render(<Menu />)

    const linkHomePage = screen.getByText('Inicial')

    expect(linkHomePage).toBeInTheDocument()
  })

  it('should render a list of links', () => {
    render(<Menu />)

    const listLinks = screen.getAllByRole('link')

    expect(listLinks).toHaveLength(4)
  })

  it('should not render an extract link', () => {
    render(<Menu />)

    const extractLink = screen.queryByText('Extrato')

    expect(extractLink).not.toBeInTheDocument()
  })

  it('should render a list of links with class link', () => {
    render(<Menu />)

    const links = screen.getAllByRole('link')

    links.forEach((link) => expect(link).toHaveClass('text-slate-900'))

    expect(links).toMatchSnapshot()
  })
})
