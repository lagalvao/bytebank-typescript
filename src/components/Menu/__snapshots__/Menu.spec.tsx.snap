// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render a list of links with class link 1`] = `
[
  <a
    class="text-slate-900 text-base cursor-pointer hover:text-black font-semibold"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="text-slate-900 text-base cursor-pointer hover:text-black font-semibold"
    href="/"
  >
    Transferências
  </a>,
  <a
    class="text-slate-900 text-base cursor-pointer hover:text-black font-semibold"
    href="/"
  >
    Investimentos
  </a>,
  <a
    class="text-slate-900 text-base cursor-pointer hover:text-black font-semibold"
    href="/"
  >
    Outros serviços
  </a>,
]
`;
